xarray/computation/computation.py

```diff
@@ -931,25 +928,30 @@ def _calc_idxminmax(
 
     # This will run argmin or argmax.
     indx = func(array, dim=dim, axis=None, keep_attrs=keep_attrs, skipna=skipna)
+    # Force dictionary format in case of single dim so that we can iterate over it in for loop below
+    if len(dims) == 1:
+        indx = {dims[0]: indx}
+
+    res = {}
+    for _dim, _da_idx in zip(dims, indx.values(), strict=False):
+        # Handle chunked arrays (e.g. dask).
+        coord = array[_dim]._variable.to_base_variable()
+        if is_chunked_array(array.data):
+            chunkmanager = get_chunked_array_type(array.data)
+            coord_array = chunkmanager.from_array(
+                array[_dim].data, chunks=((array.sizes[_dim],),)
+            )
+            coord = coord.copy(data=coord_array)
+        else:
+            coord = coord.copy(data=to_like_array(array[_dim].data, array.data))
 
-    # Handle chunked arrays (e.g. dask).
-    coord = array[dim]._variable.to_base_variable()
-    if is_chunked_array(array.data):
-        chunkmanager = get_chunked_array_type(array.data)
-        coord_array = chunkmanager.from_array(
-            array[dim].data, chunks=((array.sizes[dim],),)
-        )
-        coord = coord.copy(data=coord_array)
-    else:
-        coord = coord.copy(data=to_like_array(array[dim].data, array.data))
-
-    res = indx._replace(coord[(indx.variable,)]).rename(dim)
-
-    if skipna or (skipna is None and array.dtype.kind in na_dtypes):
-        # Put the NaN values back in after removing them
-        res = res.where(~allna, fill_value)
-
-    # Copy attributes from argmin/argmax, if any
-    res.attrs = indx.attrs
+        _res = _da_idx._replace(coord[(_da_idx.variable,)]).rename(_dim)
+        if skipna or (skipna is None and array.dtype.kind in na_dtypes):
+            # Put the NaN values back in after removing them
+            _res = _res.where(~allna, fill_value)
+        _res.attrs = _da_idx.attrs
+        res[_dim] = _res
 
+    if len(dims) == 1:
+        res = res[dims[0]]
```

Comment:
we should have some type stability here so

`idmax(dim) -> array`; `idxmax((dim,)) -> tuple[array]`; `idxmax((dim0, dim1, ...)) -> tuple[array, ...]`

Reply:
I have changed the code to match the behavior of `DataArray.arg*` which returns a dict for both idx*((dim,)) and idx*((dim0, dim1, ...))

Does that seem sensible?

Currently navigating the existing tests for arg* and multiple dims